src/Bab/RabbitMq/VhostManager.php

```diff
@@ -180,14 +180,19 @@ private function createQueues(Configuration $config)
             }
 
             for ($i = 0; $i < count($retries); $i++) {
+                if ($i === 0) {
```

Comment:
I agree, the whole `VhostManager` class is messy.
At the beginning, this project was just a simple script to save time & avoid human misconfiguration.

IMO, another solution could be to check if a queue have a retry configured in the `createBaseStructure` method. The queue will be created once & the `createQueue` method won't call the `createExchange` method anymore.

Of course, we can do the same for the `delay` logic as their is the same problem if you only use delay on some queues.

WDYT?